### PR TITLE
Use Vite to bundle vendor assets (#102)

### PR DESCRIPTION
**Pull request by mcantelon**
* Added Vite build bundling of JavaScript
* Removed unused DataTable-related JavaScript
* Updated JavaScript libraries

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,3 +17,18 @@ schema-docs:
...
 		-debug
+
+build:
+	# Install Node modules if not present
+	if [ ! -d "node_modules" ]; then \
+		npm install; \
+	fi
+
+	# Bundle vendor assets
+	vite build -c build_config/base.config.js
```
Comment (sevein) [inline Makefile:28]: We should probably use `npx vite` instead to ensure that it's using the vite installed in this environment.

File: Makefile [hunk 2]
```diff
@@ -17,3 +17,18 @@ schema-docs:
 		-cat aipscan \
 		-s aipscan \
 		-debug
+
+build:
+	# Install Node modules if not present
+	if [ ! -d "node_modules" ]; then \
+		npm install; \
+	fi
```
Comment (sevein) [inline Makefile:25]: Wouldn't you want to always run "npm install" or "npm ci"?

**Review by sevein — APPROVED** [review]
It builds really fast. I've made a couple of minor suggestions.

The size of the PR makes my browser very unhappy, hard to browse the files. There must be something going on with GitHub's UI.

File: Makefile (hunk 1 above)
Comment (mcantelon) [reply Makefile:28]: Good call!

File: Makefile (hunk 2 above)
Comment (mcantelon) [reply Makefile:25]: Good point... npm doesn't re-install what's already installed.

**Pull request merged**